src/sentry/eventstore/processing/multiredis.py

```diff
@@ -0,0 +1,72 @@
+import hashlib
+from datetime import timedelta
+from typing import TypeVar
+
+from rediscluster import RedisCluster
+
+from sentry import options
+from sentry.utils.codecs import JSONCodec
+from sentry.utils.kvstore.encoding import KVStorageCodecWrapper
+from sentry.utils.kvstore.redis import RedisKVStorage
+from sentry.utils.redis import redis_clusters
+
+from .base import EventProcessingStore
+
+T = TypeVar("T", str, bytes)
+
+
+class MultiRedisProcessingStore(EventProcessingStore):
+    """
+    Adapter to shift traffic from one redis cluster to another
+    """
+
+    def __init__(self, **options):
+        inner = MultiRedisKVStorage(
+            old_cluster=redis_clusters.get(options["old_cluster"]),
+            new_cluster=redis_clusters.get(options["new_cluster"]),
+        )
+        super().__init__(KVStorageCodecWrapper(inner, JSONCodec()))
+
+
+class MultiRedisKVStorage(RedisKVStorage[T]):
+    def __init__(self, old_cluster: RedisCluster, new_cluster: RedisCluster) -> None:
+        self.old_cluster = old_cluster
+        self.new_cluster = new_cluster
+
+    def use_new(self, key: bytes):
+        rollout = options.get("eventstore.processing.rollout")
+        intkey = int(hashlib.md5(key).hexdigest(), base=16)
+        return (intkey % 10000) / 10000 <= rollout
+
+    def get(self, key: str) -> T | None:
+        bkey = key.encode("utf8")
+        if self.use_new(bkey):
+            val = self.new_cluster.get(bkey)
+            if val is None and options.get("eventstore.processing.readold"):
+                val = self.old_cluster.get(bkey)
+            return val
+        else:
+            val = self.old_cluster.get(bkey)
+            if val is None:
+                val = self.new_cluster.get(bkey)
+            return val
+
+    def set(self, key: str, value: T, ttl: timedelta | None = None) -> None:
+        bkey = key.encode("utf8")
+        if self.use_new(bkey):
+            self.new_cluster.set(bkey, value, ex=ttl)
+        else:
+            self.old_cluster.set(bkey, value, ex=ttl)
+
+    def delete(self, key: str) -> None:
+        bkey = key.encode("utf8")
+        self.new_cluster.delete(bkey)
```

Comment:
I think you want to check the rollout before deleting from the new cluster as we did when setting the key.

Reply:
I had deletes happening on both sides to handle the scenario where we shift load to the new cluster and then have to rollback the rate for some reason. I didn't want to leave keys stranded in the new cluster.

Reply:
ok